### PR TITLE
Add protobuf example and iterate on tags.

### PR DESCRIPTION
**Pull request by gphat**
# Synopsis

This is mostly a "if this were protobuf" straw man.

* Allows tags to be a name & value or just a name.
* Makes a note about timers since they are the same as histograms.
* Adds a Protobuf schema

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,6 +34,45 @@ we use to make our systems [observable](https://en.wikipedia.org/wiki/Observabil
...
+      SET = 3;
+      EVENT = 4;
+  }
+  enum Status {
+      OK = 0;
+      WARNING = 1;
+      CRITICAL = 2;
+      UNKNOWN = 3;
+  }
+  optional Metrics metric = 1 [default = COUNTER];
```
Comment (antifuchs) [inline README.md]: seems weird that `Metrics` is the single thing that's in plural (:

File: README.md [hunk 2]
```diff
@@ -34,6 +34,45 @@ we use to make our systems [observable](https://en.wikipedia.org/wiki/Observabil
...
+  required string name = 2;
+  optional Status status = 3 [default = OK];
+  optional double value = 4;
+  optional float sample_rate = 5 [default = 1.0];
```
Comment (antifuchs) [inline README.md]: What would a sample rate of 5 mean?

File: README.md [hunk 3]
```diff
@@ -34,6 +34,45 @@ we use to make our systems [observable](https://en.wikipedia.org/wiki/Observabil
...
+syntax = "proto2";
+
+message Tag {
+  required string name = 1;
+  optional string value = 2;
+}
+
+message Trace {
+  required double id = 1;
+  optional double parent_id = 2;
```
Comment (antifuchs) [inline README.md]: I wonder if double is the right-est type for those? (Presumably, we'll want to identify things from their printed repr later on?)

File: README.md [hunk 4]
```diff
@@ -102,14 +142,13 @@ A common use of this field is as a replacement for a log line.
...
-Tags are arbitrary name-value pairs providing independent dimensions for further differentiating a metric. Such "orthogonal"
+Tags are arbitrary strings providing independent dimensions for further differentiating a metric. Such "orthogonal"
 tags are widely used and explained in other systems, so that explanation will be skipped here.
 
 SSF does not take a stance on tags meaning anything. They are just arbitrary data. Fields that have specific meaning
 to SSF are represented as properties in this spec, rather than in tags.
 
-The field `tags`, if present, must contain an object with an arbitrary number of fields. Each field and it's value
-must be a string.
+The field `tags`, if present, must contain an array with an arbitrary number of strings.
```
Comment (antifuchs) [inline README.md]: I think with the tags pb defn above (which I suuuper love), we should adjust that statement (:

File: README.md (hunk 2 above)
Comment (gphat) [inline README.md]: It'd mean 1.0? 😺 Protobuf can't set acceptable ranges for a number, afaik. What could I do here?

Comment (antifuchs) [inline README.md]: Unclear - we could define it as a percent-based range, too: 0-100 (but then, what is 101? What's 10001?). Guess none of the options we have available really work well - but I think the example we give should make sense, at least; make it `0.75` or something?

Comment (gphat) [inline README.md]: Wait, do you mean the literal 5? That's its position in the spec. The
default is 1.0 and we'd have a separate example. Or did I misunderstand?

**Comment by gphat** [comment]
This needs a span id added to be compatible with trace implementations, fwiw.

**Pull request merged**